source/NVDAObjects/window/winword.py

```diff
@@ -426,8 +426,16 @@ def iterate(self):
 			if self.direction=="previous":
 				index=itemCount-(index-1)
 			collectionItem=items[index]
-			item=self.quickNavItemClass(self.itemType,self.document,collectionItem)
-			itemRange=item.rangeObj
+			try:
+				item=self.quickNavItemClass(self.itemType,self.document,collectionItem)
+				itemRange=item.rangeObj
+			except:
+				values = (self.itemType, self.direction, itemCount, index)
+				message = ("Error iterating over item with "+
+				("type: %s, iteration direction: %s, total item count: %s, item at index: %s" % values )+
+				"\nThis could be caused by an issue with some element within or a corruption of the word document.")

```

Comment:
1. This would probably be easier to read using Python's str.format function.
2. If you're building a big string, you can just put multiple string literals on multiple lines (inside parentheses) and Python will treat them as a single literal. This is just syntactic sugar.

Applying both of the above would look like this:

```
message = ("Error iterating over item with "
	"type: {type}, iteration direction: {direction}, total item count: {count}, item at index: {index}\n"
	"This could be caused by an issue with some element within or a corruption of the word document.").format(
	type=self.itemType, direction=self.direction, count=itemCount, index=index)
```